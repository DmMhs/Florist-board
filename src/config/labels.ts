const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
