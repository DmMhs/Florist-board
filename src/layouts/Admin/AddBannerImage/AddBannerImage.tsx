import React, { Component } from 'react';

import { AppContext } from '../../../AppContext';
import { storageRef, homeImagesRef } from '../../../firebase';
import './AddBannerImage.less';
import { uploadBannerImage } from '../../../services/admin/uploadBannerImage';
import { getBannerImageDownloadURL } from '../../../services/admin/getBannerImageDownloadURL';

interface AddGalleryImageProps {}

interface AddGalleryImageState {
  images: File[];
  totalImagesNumber: number;
}

class AddBannerImage extends Component<
  AddGalleryImageProps,
  AddGalleryImageState
> {
  constructor(props: AddGalleryImageProps) {
    super(props);
    this.state = {
      images: [],
      totalImagesNumber: 0
    };
  }

  public componentDidMount() {
    homeImagesRef.on('value', snapshot => {
      this.setState({
        totalImagesNumber: Object.keys(snapshot!.val()).length
      });
    });
  }

  private imgInputChangedHandler = (
    position: number,
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    const updatedImages: File[] = [...this.state.images];
    updatedImages[position] = event.target.files as any;
    this.setState({
      images: updatedImages
    });
  };

  private formSubmitHandler = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const { images } = this.state;
    this.setState({
      images: []
    });
    images.map(async (image: File) => {
      const file = (image as any)[0];
      const formattedFileName = (image as any)[0].name;
      await uploadBannerImage(file, formattedFileName);
      getBannerImageDownloadURL(formattedFileName)
        .then(res => {
          return res;
        })
        .then(imgURL => {
          homeImagesRef.push(imgURL);
        });
    });
  };

  public render() {
    const context = this.context;
    const labels = context.state.labels;
    const lang = context.state.lang;

    const labelsRoot = labels[lang].pages.admin.addBannerImageForm;
    const submitBtnLabel = labels[lang].pages.admin.submitBtn;

    return (
      <AppContext.Consumer>
        {value =>
          value && (
            <form
              onSubmit={this.formSubmitHandler}
              className="AddGalleryImage form"
            >
              <div className="form-control product-images">
                <p className="gallery-info">
                  {labelsRoot.info.totalImagesNumber}{' '}
                  {this.state.totalImagesNumber}
                </p>
                <hr />
                <label className="accent">{labelsRoot.add}</label>
                <br />
                <div className="input-wrapper">
                  <input
                    type="file"
                    onChange={this.imgInputChangedHandler.bind(this, 0)}
                  />
                </div>
              </div>
              <button type="submit">{submitBtnLabel}</button>
            </form>
          )
        }
      </AppContext.Consumer>
    );
  }
}

AddBannerImage.contextType = AppContext;

export default AddBannerImage;
