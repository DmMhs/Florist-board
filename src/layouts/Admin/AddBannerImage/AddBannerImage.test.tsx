import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import AddBannerImage from './AddBannerImage';
import AppContextProvider from '../../../AppContext';
import labels from '../../../config/labels';

describe('AddBannerImage works as expected', () => {
  it('AddBannerImage component matches a snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <AddBannerImage />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
