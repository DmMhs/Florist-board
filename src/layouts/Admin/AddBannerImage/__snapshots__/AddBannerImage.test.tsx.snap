// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBannerImage works as expected AddBannerImage component matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <AddBannerImage>
            <form
              className="AddGalleryImage form"
              onSubmit={[Function]}
            >
              <div
                className="form-control product-images"
              >
                <p
                  className="gallery-info"
                >
                  Number of banner images: 
                   
                  0
                </p>
                <hr />
                <label
                  className="accent"
                >
                  Add new banner image:
                </label>
                <br />
                <div
                  className="input-wrapper"
                >
                  <input
                    onChange={[Function]}
                    type="file"
                  />
                </div>
              </div>
              <button
                type="submit"
              >
                SUBMIT
              </button>
            </form>
          </AddBannerImage>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
